vllm/model_executor/guided_decoding/xgrammar_decoding.py

```diff
@@ -169,19 +187,17 @@ def from_guided_params(cls,
                 grammar_str = guided_params.grammar
             return cls(grammar_str=grammar_str,
                        vocab_size=model_config.hf_text_config.vocab_size,
-                       encoded_vocab=encoded_vocab,
-                       stop_token_ids=stop_token_ids,
-                       backend_str=backend_str,
                        tokenizer_hash=tokenizer_hash,
-                       max_threads=max_threads)
+                       max_threads=max_threads,
+                       tokenizer_data=tokenizer_data)
         elif guided_params.json_object:
-            return cls(json_object=True,
-                       vocab_size=model_config.hf_text_config.vocab_size,
-                       encoded_vocab=encoded_vocab,
-                       stop_token_ids=stop_token_ids,
-                       backend_str=backend_str,
-                       tokenizer_hash=tokenizer_hash,
-                       max_threads=max_threads)
+            return cls(
+                json_object=True,
+                vocab_size=model_config.hf_text_config.vocab_size,
+                tokenizer_hash=tokenizer_hash,
+                max_threads=max_threads,
+                tokenizer_data=tokenizer_data,
```

Comment:
can you check if xgr.VocabType is serializable with pickle?

the reason for populate each of the field due to the fact that some object in xgrammar is not serialisable yet.

Reply:
See L218 for `__getstate__` and `__setstate__` of the config

Reply:
I confirm that xgr.VocabType is serializable, added a test that check that.